cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramRuntimeProvider.java

```diff
@@ -47,7 +47,7 @@
    * The execution mode of the program runtime system.
    */
   enum Mode {
-    LOCAL, DISTRIBUTED
+    IN_MEMORY, LOCAL, DISTRIBUTED
```

Comment:
I added this so that there was a clearer distinction between the three ProgramRunners now. `IN_MEMORY` mode returns a program runner that will be used to create the LOCAL mode program runner. `IN_MEMORY` mode was also needed to wrap the state change listener around the controller. 

This is analogous to `DISTRIBUTED` mode returning a program runner that will create the `LOCAL` mode program runner.